### PR TITLE
Swap glamorous for styled components

### LINKED ISSUE #51: Replace glamorous with styled-components
[`swap-glamorous-for-styled-components`](https://github.com/iFixit/toolbox/compare/swap-glamorous-for-styled-components)
## Problem
 > We have since backed the other horse in this race and placed many bets on it. Let's take glamorous behind the barn and put it down.

 > style-components is the golden cache horse.

Glamorous has been deprecated, and we have standardized on styled-components. Since it's based on some deprecated features of React, this is blocking us from upgrading React.

## Do This
Replace all the use cases of Glamorous in this repo with styled-components and drop the dependency. This includes any updates to our jest tests.

Connects https://github.com/iFixit/ifixit/issues/31530

### PR DESCRIPTION
**Pull request by sterlinghirsh**
Glamorous is deprecated. The swap was pretty straightforward.

QA: Check the docz sites on master and this branch side by size. I updated all the snapshot tests so you can't trust them.
This may be easiest with https://github.com/iFixit/jux

Closes #51

### CODE REVIEW

**Comment by zdmitchell** [comment]
CR :guitar:

File: src/components/SearchInput/SearchInput.jsx [hunk 1]
```diff
@@ -69,13 +69,17 @@ const Input = glamorous.input(
       },
 );
 
+const RelativeDiv = styled.div`
+   position: relative;
+`;
+
 const SearchInput = ({ className, ...props }) => (
-   <Div className={className} position="relative">
+   <RelativeDiv className={className} position="relative">
```
Comment (andrewjpiro) [inline src/components/SearchInput/SearchInput.jsx:77]: [minor] The `position="relative"` on this line is extraneous.

**Comment by andrewjpiro** [comment]
CR :boom:

**Comment by ardelato** [comment]
QA 🍔 

## Steps taken
1. Cloned the toolbox repo twice locally
    1a. One to a Folder for the `master` branch
    1b. One to a Folder for the `swap-glamorous-for-styled-components` branch

2. Cloned the [jux repo](https://github.com/iFixit/jux) to easily compare and contrast both the `master` and `swap-glamorous-for-styled-components` branch

3. Ran `npm install` with each folder directory:
    3a. toolbox `master`
    3b. toolbox ``swap-glamorous-for-styled-components`
    3c. jux

4. Ran `npm start` on two different terminal instances to start the servers for each branch
    4a. One for the `master` branch at `localhost:3001`
    4b. One for ``swap-glamorous-for-styled-components` branch at `localhost:3000` 

5. Modified the `jux`'s `src/pages.json` file to have the `master` branch as the `default_comparison_source` and the `swap-glamorous-for-styled-components` branch as the `comparison_target. While also added the pages for all the changed Compenents to the `pages` array.
```json
{
  "pages": ["", "components/avatar", "components/button", "components/button-group", "components/checkbox","components/icon", "components/label", "components/pill-badge", "components/radio-group", "components/search-input", "components/tab", "components/tabs", "components/text-field", "components/textarea"],
  "default_comparison_source": "localhost:3001",
  "comparison_target": "localhost:3000",
  "deploy_target": "rsync:/server/to/deploy/to"
}
```

6. I also modified `jux`'s `src/comparison_urls.js` to change the format url from `https` -> `http` as it wasn't loading the locally hosted websites
```diff
--- a/src/comparison_urls.js
+++ b/src/comparison_urls.js
@@ -31,7 +31,7 @@ function splitUrl(url) {
 }
 
 function formatUrl(host, path) {
-  return `https://${host}/${path}`;
+  return `http://${host}/${path}`;
 }
 ```

 7. I then went through the pages array and verified that the components for `swap-glamorous-for-styled-components` branch matched those of the `master` branch
    [Jump Share Folder link for each component comparison](https://jmp.sh/rShxZdI)



### Some issues
There is currently no `.mdx` files for the `Components`: `Grid, PageNavBar, Tag` and for the `Constant`: `animation`. 

I was able to create `.mdx` files for `Grid` and `PageNavBar` for testing but was not able to do the same for `Tag` and `animation`. Aside from those two, everything else that had a change was tested and looked good.

**Pull request merged**